src/scripts/main.js

```diff
@@ -1,3 +1,31 @@
 'use strict';
 
-// write code here
+const employeesContainer = document.querySelector('ul');
+const employees = [...document.querySelectorAll('[data-salary]')];
+const employeesList = employees.map((employee) => {
+  return {
+    name: employee.innerText,
+    position: employee.dataset.position,
+    salary: +employee.dataset.salary.replace(/[$,]/g, ''),
+    age: employee.dataset.age,
+  };
+});
+
+const sortEmployees = employeesList.sort((a, b) => b.salary - a.salary);
```

Comment:
The 'sortEmployees' variable is currently being assigned the result of sorting 'employeesList'. Be aware that the 'sort' method sorts the array in place, meaning the original 'employeesList' array will be sorted as well. If you want to keep a separate sorted list, consider using the 'slice' method to create a copy of 'employeesList' before sorting.